### PR TITLE
Add possibility to override connection string with a strongly typed options

### PR DESCRIPTION
**Pull request by DmyMi**
- Base options needed for every service
- Possible to have different options per service
- Possible to change any database connection option by changing a single environment variable instead of generating a new connection string all the time even for a small change

### CODE REVIEW

File: OutOfSchool/OutOfSchool.Common/Config/IMySqlConnectionOptions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+namespace OutOfSchool.Common.Config
+{
+    public interface IMySqlConnectionOptions
+    {
+        public bool UseOverride { get; set; }
+
+        public string Server { get; set; }
+
+        public uint Port { get; set; }
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.Common/Config/IMySqlConnectionOptions.cs:9]: Wow `uint`! First time see this type really used in C# project :)

Comment (DmyMi) [reply OutOfSchool/OutOfSchool.Common/Config/IMySqlConnectionOptions.cs:9]: it is used in the Pomelo `MySqlConnectionBuilder` class :(

File: OutOfSchool/OutOfSchool.Common/Extensions/Startup/ConnectionStringExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    ConnectionString = config.GetConnectionString(connectionStringKey),
+                };
+            }
+
+            if (string.IsNullOrEmpty(connectionStringBuilder.ConnectionString))
+            {
+                throw new Exception("Provide a valid connection string or options");
+            }
+
+            if (!connectionStringBuilder.ContainsKey("guidformat") || connectionStringBuilder["guidformat"].ToString().ToLower() != "binary16")
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.Common/Extensions/Startup/ConnectionStringExtensions.cs:46]: Please don't use `.ToLower()` for comparing strings, please use `String.Equals(..., ..., StringComparison.OrdinalIgnoreCase)`

**Review by VadymLevkovskyi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.Common/Config/IMySqlConnectionOptions.cs (hunk 1 above)
Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.Common/Config/IMySqlConnectionOptions.cs:9]: Well, then all credits go to Pomelo author.

**Pull request merged**